core/src/main/java/com/backblaze/b2/client/B2StorageClient.java

```diff
@@ -237,6 +238,7 @@ B2FileVersion uploadLargeFile(B2UploadFileRequest request,
      * @param fileVersion The B2FileVersion for the large file getting stored.
      *                    This is the return value of startLargeFile().
      * @param contentSource The contentSource to upload.
+     * @param uploadListener The object that handles upload progress events.
```

Comment:
can you look at other places we let you provide upload listeners?  does this follow our style?  in particular, should we allow null (and convert it to a noopListener internally?)

Reply:
Providing that API popped into my head. Since you suggested it too I can take care of that. For uploading a large file, the upload listener is provided via the `B2UploadFileRequest` which is passed to the storage client. I figured this was the closest thing. Your thoughts?

Reply:
B2UploadFileRequest's constructor says this:
        this.listener = (listener != null) ? listener : B2UploadListener.noopListener();

and the builder for that structure defaults to null, so i'm inclined to let people pass null in your case.

and this discussion makes me think about the fact that many of the methods on the storage client take a request structure that is build with a builder.  this lets us add new options without causing binary or source-level incompatibilities.

Reply:
Are you suggesting creating a StoreLargeFile request with a corresponding builder?